main.go

```diff
@@ -16,17 +19,49 @@ func proxyHandler(w http.ResponseWriter, r *http.Request) {
 		"header":     r.Header,
 	}).Info("Received /proxy request")
 
+	target := r.Header.Get("X-OXYproxy-target")
```

Comment:
you almost always should check user input
`target` can be empty, lets handle this case